core/src/main/scala/shapeless/labelled.scala

```diff
@@ -79,7 +79,9 @@ trait FieldPoly extends Poly1 {
 trait FieldOf[V] {
   import labelled._
 
-  def ->>(v: V): FieldType[this.type, V] = field[this.type](v)
+  type FieldType = labelled.FieldType[this.type, V]
+
+  def ->>(v: V): labelled.FieldType[this.type, V] = field[this.type](v)
```

Comment:
Interesting. I don't see any obvious reason for that.

It's also annoying that you have to qualify `FieldType` here. Maybe it would be better to go with `type F`? It's being used primarily as an abbreviation, so a single character name would be justified I think.

If you do that then I think the original definition of `->>` should be fine.